content/docs/configuration-services.md

```diff
@@ -0,0 +1,327 @@
+---
+title: "Configuration via Services"
+---
+Langium supports the configuration of most aspects of your language and language server via a set of `Services`. Those services are organized into `Modules`, which are essentially mappings from a service name to its implementation.
+
+We can separate modules into two main categories:
+1. ***Shared modules*** which contain services that are shared across all languages.
+2. ***Language specific modules*** which contain services specific to one language. 
+
+## Shared Modules
+The *shared modules* contain services shared across all languages:
+* The `ServiceRegistry` is responsible for registering and accessing the different services.
+* The `Connection` service holds information about the connection between client and server.
+* Services to handle `Langium Documents` and `Text Documents`.
+* Implementations of the `AstReflection` service (one implementation per language) to access the structure of the AST.
+
+## Language Specific Modules
+The *language specific modules* contain all services which are specific to one language:
+* Services to build the `parser` for a specific language.
+* Services to handle `LSP requests`.
+* Services to retrieve information from a given AST.
+* Services to manage references during linking and inside of documents.
+* Serialization and deserialization of objects to JSON.
+* Services to handle document and syntax validation.
+
+## Customization
+The entry point to services customization is found in the `src/language-server/...-module.ts` file, where '...' is the name of your language. There, you can register new services, or override the default implementation of services. Langium implements the *Inversion of Control* pattern via *Dependency Injection*, which promotes loosely-coupled architectures, maintainability, and extensibility.
```

Comment:
How I understand IoC is a principle and DI is a pattern, so: "Langium implements the Inversion of Control principle via the Dependency Injection pattern".

Reply:
I will change the sentence